tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousTest.java

```diff
@@ -65,34 +70,21 @@ public static WebArchive deploy() {
      */
     @Test
     public void testAsyncIsNotFinished() {
-        Future<Connection> future = null;
-        try {
-            future = client.service();
-        }
-        catch (InterruptedException e) {
-            throw new AssertionError("testAsync: unexpected InterruptedException calling service");
-        }
-
-        Assert.assertFalse(future.isDone());
-
+        CompletableFuture<Void> waitingFuture = newWaitingFuture();
+        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture);
+        Future<Connection> future = resultFuture.toCompletableFuture();
+        await().atMost(200, TimeUnit.MILLISECONDS).untilAsserted(()-> Assert.assertFalse(future.isDone()));
```

Comment:
Assuming there is a `service` method taking `waitingFuture` and returning `Future`, I think this test should look roughly like this:

```java
CompletableFuture<Void> waitingFuture = newWaitingFuture();
Future<Connection> future = client.service(waitingFuture);
assertFalse(future.isDone());
```

There's no need for awaiting the `!isDone()` state.

Reply:
done